php-frontend/src/main/java/org/sonar/php/parser/NewPHPGrammar.java

```diff
@@ -0,0 +1,1045 @@
+          b.oneOrMore(
+              b.firstOf(
+                  ENCAPSULATED_STRING_VARIABLE(),
+                  EXPANDABLE_STRING_CHARACTERS()
+              )
+          ),
+          b.token(PHPLexicalGrammar.DOUBLE_QUOTE)
+      ));
+  }
+
+  public ExpressionTree ENCAPSULATED_STRING_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.ENCAPS_VAR)
+      .is(b.firstOf(
+        ENCAPSULATED_SEMI_COMPLEX_VARIABLE(),
+        ENCAPSULATED_SIMPLE_VARIABLE(),
+        ENCAPSULATED_COMPLEX_VARIABLE()
+        )
+      );
+  }
+
+  public ExpressionTree ENCAPSULATED_COMPLEX_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.COMPLEX_ENCAPS_VARIABLE)
+      .is(f.encapsulatedComplexVariable(
+          b.token(LCURLYBRACE),
+          b.token(PHPLexicalGrammar.NEXT_IS_DOLLAR),
+          EXPRESSION(),
+          b.token(RCURLYBRACE)
+      ));
+  }
+
+  public ExpressionTree ENCAPSULATED_SEMI_COMPLEX_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.SEMI_COMPLEX_ENCAPS_VARIABLE)
+      .is(f.encapsulatedSemiComplexVariable(
+          b.token(DOLLAR_LCURLY),
+          b.firstOf(
+              EXPRESSION(),
+              f.expressionRecovery(b.token(PHPLexicalGrammar.SEMI_COMPLEX_RECOVERY_EXPRESSION))),
+          b.token(RCURLYBRACE)));
+  }
+
+  public ExpressionTree ENCAPSULATED_SIMPLE_VARIABLE() {
+    return b.<ExpressionTree>nonterminal(PHPLexicalGrammar.SIMPLE_ENCAPS_VARIABLE)
+      .is(
+        f.encapsulatedSimpleVar(ENCAPSULATED_VARIABLE_IDENTIFIER(),
+            b.optional(b.firstOf(
+                f.expandableArrayAccess(
+                    b.token(LBRACKET),
+                    b.firstOf(IDENTIFIER(), NUMERIC_LITERAL(), ENCAPSULATED_VARIABLE_IDENTIFIER()),
+                    b.token(RBRACKET)),
+                f.expandableObjectMemberAccess(b.token(ARROW), IDENTIFIER()))))
+      );
+  }
+
+  public IdentifierTree IDENTIFIER() {
+    return b.<IdentifierTree>nonterminal(Kind.IDENTIFIER)
+      .is(f.identifier(b.token(PHPLexicalGrammar.IDENTIFIER)));
+  }
+
+  public VariableIdentifierTree ENCAPSULATED_VARIABLE_IDENTIFIER() {
+    return b.<VariableIdentifierTree>nonterminal(PHPLexicalGrammar.ENCAPS_VAR_IDENTIFIER)
+      // variable identifiers encapsulated into strings literal does not allowed line terminator spacing,
+      // so here using "WHITESPACE" instead of SPACING.
+      .is(f.encapsulatedVariableIdentifier(b.token(PHPLexicalGrammar.WHITESPACES), b.token(PHPLexicalGrammar.VARIABLE_IDENTIFIER)));
+  }
+
+  public ExpressionTree EXPANDABLE_STRING_CHARACTERS() {
+    return b.<ExpandableStringCharactersTree>nonterminal(Kind.EXPANDABLE_STRING_CHARACTERS)
+      .is(f.expandableStringCharacters(b.token(PHPLexicalGrammar.STRING_WITH_ENCAPS_VAR_CHARACTERS)));
+  }
+
+  public YieldExpressionTree YIELD_EXPRESSION() {
+    return b.<YieldExpressionTree>nonterminal(Kind.YIELD_EXPRESSION)
+      .is(f.yieldExpression(
+          b.token(YIELD),
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Take the required action to fix the issue indicated by this comment. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1134)